doc/design/optimizer.md

```diff
@@ -0,0 +1,85 @@
+## Optimizer Design
+In deeplearning system, `Optimizer` is used to optimize(minimize) loss thow updating a list of parameters. 
+
+### A typical training process:
```

Comment:
If the above proposed section `## The Problem` is accepted, this paragraph of three bullets can be removed.

Reply:
done